src/main/java/pet/odyvanck/petclinic/service/OwnerServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package pet.odyvanck.petclinic.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pet.odyvanck.petclinic.dao.OwnerRepository;
+import pet.odyvanck.petclinic.domain.Owner;
+import pet.odyvanck.petclinic.domain.User;
+import pet.odyvanck.petclinic.domain.error.EntityNotFoundException;
+import pet.odyvanck.petclinic.service.specification.OwnerSpecification;
+import pet.odyvanck.petclinic.web.dto.owner.OwnerRequestParams;
+import pet.odyvanck.petclinic.web.dto.owner.OwnerUpdateRequest;
+
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+
+@Service
+@RequiredArgsConstructor
+public class OwnerServiceImpl implements OwnerService {
+
+    private final UserService userService;
+    private final OwnerRepository ownerRepository;
+
+    @Transactional
+    @Override
+    public Owner register(Owner owner, User user, String password) {
```

Comment:
В дополнение к notnull, nullable. Можно на поля что notnull вещать requireNonNull. В Яндексе мы активно использовали эту практику  
```suggestion
    public Owner register(Owner owner, User user, String password) {
          Objects.requireNonNull(owner, "owner must be not null");
```